R/md_document.R

```diff
@@ -73,18 +73,31 @@ md_document <- function(variant = "markdown_strict",
     }
   }
 
-  # return format
-  output_format(
-    knitr = knitr_options_html(fig_width, fig_height, fig_retina, FALSE, dev),
-    pandoc = pandoc_options(
-      to = variant,
-      from = from_rmarkdown(extensions = md_extensions),
-      args = args,
-      ext = ext,
-      lua_filters = if (number_sections) pkg_file_lua("number-sections.lua")
-    ),
-    clean_supporting = FALSE,
-    df_print = df_print,
-    post_processor = post_processor
-  )
+    output_format(
+      knitr = knitr_options_html(fig_width, fig_height, fig_retina, FALSE, dev),
+      pandoc = pandoc_options(
+        to = variant,
+        from = from_rmarkdown(extensions = md_extensions),
+        args = args,
+        ext = ext,
+        lua_filters = if (number_sections && ext != ".gfm") pkg_file_lua("number-sections.lua")
```

Comment:
As I already mentioned, flavors other than `gfm` also require `number_sections` to be preprocessed.
I would like to consider about it further in #2093.
Thus, I would like to ask you keep `output_format` unchanged, and make your PR to be limited to file extensions.
Is this fine for you?

https://github.com/rstudio/rmarkdown/pull/2094#pullrequestreview-627432449

Reply:
> As I already mentioned, flavors other than `gfm` also require `number_sections` to be preprocessed.
> I would like to consider about it further in #2093.
> Thus, I would like to ask you keep `output_format` unchanged, and make your PR to be limited to file extensions.
> Is this fine for you?
> 
> [#2094 (review)](https://github.com/rstudio/rmarkdown/pull/2094#pullrequestreview-627432449)

Hello, see if the PR is in the form you want.